src/modules/client/components/clientAlerts/CreateClientAlertDialog.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  Alert,
+  Box,
+  DialogContent,
+  DialogProps,
+  DialogTitle,
+} from '@mui/material';
+import CommonDialog from '@/components/elements/CommonDialog';
+import StaticForm from '@/modules/form/components/StaticForm';
+import { clientBriefName } from '@/modules/hmis/hmisUtil';
+import { cache } from '@/providers/apolloClient';
+import {
+  ClientAlertInput,
+  ClientNameFragment,
+  CreateClientAlertDocument,
+  CreateClientAlertMutation,
+  CreateClientAlertMutationVariables,
+  StaticFormRole,
+} from '@/types/gqlTypes';
+
+interface ClientAlertDialogProps extends DialogProps {
+  client: ClientNameFragment;
+}
+export const CreateClientAlertDialog: React.FC<ClientAlertDialogProps> = ({
+  client,
+  ...props
+}) => {
+  const clientName = clientBriefName(client);
+
+  return (
+    <CommonDialog maxWidth='sm' fullWidth {...props}>
+      <DialogTitle>Create Alert for {clientName}</DialogTitle>
+      <DialogContent>
+        <Box mt={2}>
+          <Alert sx={{ mb: 2 }} severity='info'>
+            Reminder: All staff who can view {clientName} will be able to see
+            this alert
+          </Alert>
+          <StaticForm<
+            CreateClientAlertMutation,
+            CreateClientAlertMutationVariables
+          >
+            role={StaticFormRole.ClientAlert}
+            mutationDocument={CreateClientAlertDocument}
+            getVariables={(values) => {
+              return {
+                input: {
+                  clientId: client.id,
+                  ...values,
+                } as ClientAlertInput,
+              };
+            }}
+            getErrors={(data) => data.createClientAlert?.errors || []}
+            onCompleted={(data) => {
```

Comment:
Somehow I must still be missing something related to getting required fields working properly -- they now do appear with the little 'Required' red label, but on submitting the form, it still makes a network call (instead of erroring because the form's invalid, like the Auto Exit Config form does).

Reply:
Ah, my bad. I was thinking of the "record editing forms" which all use the `SubmitForm` mutation. For _those_ types of forms, it's sufficient to specify requiredness in the FormDefinition because the mutation actually validates against the form ([code](https://github.com/greenriver/hmis-warehouse/blob/256a2a28946c1e773c17d76fe0198bcd6f8afbff/drivers/hmis/app/graphql/mutations/submit_form.rb#L70)).

Since this is a static form that uses a custom mutation, it's a little different. I think you'll need the following changes:
* Make `note` input optional on the `ClientAlertInput`. This mutation is accepting the user input directly, so we need to  allow that to be empty (or else GraphQL will reject the request)
* Validate input in the mutation:
      ```
      errors = HmisErrors::Errors.new
      errors.add :note, :required if input.note.blank?
      errors.add :priority, :required unless input.priority
      return { errors: errors } if errors.any?
      ```


That should let you see user-facing validation when you submit with those fields empty. It does make a server request.

Reply:
cool! Thank you.